src/include/migraphx/pre_scheduling.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_PRE_SCHEDULING_HPP
+#define MIGRAPHX_GUARD_RTGLIB_PRE_SCHEDULING_HPP
+
+#include <string>
+#include <migraphx/instruction_ref.hpp>
+#include <migraphx/operation.hpp>
+
+#include <migraphx/program.hpp>
+#include <migraphx/insert_instruction.hpp>
+namespace migraphx {
+
+struct pre_scheduling
+{
+    std::function<std::pair<int, int>(const operation&)> weight_func;
```

Comment:
Can we make the function return a struct instead of a pair?

Reply:
I tried to make it as generic as possible now.  When there is a need to enhance machine models, we will make more changes.

Reply:
I dont understand this. Why cant this be a struct with more meaningful names than `first` and `second`?

Reply:
I mean not using struct, keep it as std::pair for now.   Things like this can be delayed when there is a need in the future.  I have spent too much time in this kind of coding style changes that do not bring any values.

Reply:
> I have spent too much time in this kind of coding style changes that do not bring any values.

It does bring a lot of value. It improves readability and makes it clearer what each integer is supposed to return.

I dont recall what each integer is, and would need to look at notes from previous discussion instead of the source code to find out. Adding a name to them will help a lot.

Reply:
When there are fixed guidelines of coding styles that teams agree on, I will conform to it.  But you are the sole gatekeeper to dictate whatever code should be look like is unproductive.